src/transformers/pipelines/image_to_text.py

```diff
@@ -69,76 +71,37 @@ def _sanitize_parameters(self, max_new_tokens=None, generate_kwargs=None):
                     " please use only one"
                 )
             forward_kwargs["generate_kwargs"]["max_new_tokens"] = max_new_tokens
+        if texts is not None:
+            preprocess_params["texts"] = texts
```

Comment:
NIT: Let's keep preprocess before forward before postprocess (just for reader's sanity)

Reply:
Should be fixed in [7fc991f](https://github.com/huggingface/transformers/pull/22423/commits/7fc991f738691fff13626688e94df597e850decc)!